lib/routes/sohu/mobile.ts

```diff
@@ -0,0 +1,209 @@
+    const jsonMatch = jsonScript?.match(/window\.WapHomeRenderData\s*=\s*({.*})/s);
+    if (!jsonMatch?.[1]) {
+        throw new Error('WapHomeRenderData 数据未找到');
+    }
+    const renderData = JSON.parse(jsonMatch[1]);
+    const list = extractPlateBlockNewsLists(renderData)
+        .filter((item) => item.id && item.url?.startsWith('//'))
+        .map((item) => ({
+            title: item.title,
+            link: new URL(item.url.split('?')[0], 'https://m.sohu.com').href,
+        }));
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                try {
+                    const detailResp = await ofetch(item.link);
+                    const $d = cheerio.load(detailResp);
+                    return {
+                        ...item,
+                        description: getDescription($d) || item.title,
+                        pubDate: extractPubDate($d),
+                    };
+                } catch (error) {
+                    logger.error(`获取详情失败: ${item.link}`, error);
+                    return item;
+                }
+            })
+        )
+    );
+    return {
+        title: '手机搜狐新闻',
+        link: 'https://m.sohu.com/limit',
+        item: items.filter(Boolean),
+    };
+}
+
+function extractPlateBlockNewsLists(jsonData: any) {
+    const result: any[] = [];
+    for (const key of Object.keys(jsonData)) {
+        if (key.startsWith('PlateBlock')) {
+            const plateBlock = jsonData[key];
+            // 处理新闻列表
+            if (plateBlock?.param?.newsData?.list) {
+                result.push(...plateBlock.param.newsData.list);
+            }
+            // 处理焦点图数据
+            if (plateBlock?.param?.focusData?.list) {
+                result.push(...plateBlock.param.focusData.list);
+            }
+            if (plateBlock?.param?.feedData0?.list) {
+                result.push(...plateBlock.param.feedData0.list);
+            }
+            if (plateBlock?.param?.feedData1?.list) {
+                result.push(...plateBlock.param.feedData1.list);
+            }
+        }
+    }
+    return result;
+}
+
+function extractPubDate($: cheerio.CheerioAPI): string {
+    const timeElements = [
+        '.time',
+        '#videoPublicTime',
+        'div.channelHeaderContainer div.detail',
+    ];
+    for (const selector of timeElements) {
+        const text = $(selector).first().text().trim();
+        if (!text) {
+            continue;
+        }
+        //  ‘奔流 · 昨天23:59 · 2074阅读’
+        const dateRegex = /\d{4}[-/]\d{2}[-/]\d{2} \d{2}:\d{2}|昨天\d{1,2}:\d{2}|\d+\s*(?:小时|分钟)前|\d{2}-\d{2} \d{2}:\d{2}|今天\d{1,2}:\d{2}|\d{4}年\d{2}月\d{2}日 \d{2}:\d{2}/;
+        const dateMatch = text.match(dateRegex);
+        const date = dateMatch?.[0];
+        if (date) {
+            const newdate = parseDate(date);
+            if (newdate) {
+                return newdate;
+            }
+        }
+    }
+    return new Date().toISOString();
```

Comment:
```suggestion
    const timeElements = ['.time', '#videoPublicTime'];
    let date;
    for (const selector of timeElements) {
        const text = $(selector).first().text().trim();
        if (!text) {
            continue;
        }

        date = parseDate(text);
        if (date) {
            return date;
        }
    }

    const img = $('meta[name="share_img"]')
        .toArray()
        .map((i) => $(i).attr('src'))
        .find((i) => i.includes('images01'));
    date = img ? parseDate(img?.match(/images01\/(\d{8})\//i)?.[1]) : '';
    if (date) {
        return date;
    }
```
[Do not add](https://docs.rsshub.app/joinus/advanced/pub-date#no-date) a date if the site does not provide one.